### DISCUSSION TITLE
Using a custom model created in birdnet analyser  in a BirdNET-PI station

### DISCUSSION THREAD

**Discussion by gutocarv** [Nachtzuster/BirdNET-Pi#15] [Q&A]
Hi all 
We are working on monitoring the Cherry-throated-tanager (Nemosia rourei), a highly endangered species, in the forests of Southeast Brazil. We have experimentally installed a BirdNET-PI station, and the results have been very promising for numerous species. However, the current BirdNET model does not identify our target species, the Cherry-throated-tanager (Nemosia rourei).
So, we created a model using [BirdNET Analyzer](https://github.com/kahst/BirdNET-Analyzer) to use it in the BirdNETPI station,  but we couldn't get it to work.

Question: Is it possible to use a model created in BirdNET Analyzer on BirdNETPI?
If yes, how can we do this? What parameters need to be adjusted?"

Our team is not familiar with model creation or the use of TensorFlow, but we managed to create a highly accurate model using BirdNET Analyzer. It would be amazing to be able to use it in the field to monitor the Cherry-throated Tanager.

Any help will be higly appreciated

**Comment by Nachtzuster** [comment]
Interesting.
BirdNET-Pi is using [this](https://github.com/kahst/BirdNET-Analyzer/blob/main/checkpoints/V2.4/BirdNET_GLOBAL_6K_V2.4_Model_FP16.tflite) model, so I assume this should indeed be doable.

I would re-post the question in the original BirdNET-Pi and Birdnet Analyser too, I'm sure it would get way more views than here :smiley:

**Reply by simonlinke** [reply]
Do you reckon that we could just drop a .tflite classifier in the model folder of BirdNetPi? 
This is not batty - it's a normal 48k classifier

**Reply by simonlinke** [reply]
Actually, I'll just answer my own question - no you can't haha. This would be really great to add custom tflites. Unfortunately I'm lacking those skills (but maybe I'll try rdz-oss's approach)

**Comment by rdz-oss** [comment]
Hi,

you can build a bridge from 

- the server.py on BirdNET-Pi to the
- server.py on the BirdNET-Analyzer

the BirdNET-Analyzer can work with a custom classifier.  So instead of using a local BirdNET model you can use any classifier you traine via BirdNET-Analyzer.

I did this to get BattyBirdNET-Pi to work with BattyBirdNET-Analyzer. Had to add quite a few more mods to get the system to work in ultra sound 256kHZ instead of the 48kHz of BirdNET-Pi.
But you can build such a bridge also for the regular BirdNET-Pi by forking it or convincing the maintainer to do it.

This is how the bridge works:
from BattyBirdNET-Pi
https://github.com/rdz-oss/BattyBirdNET-Pi/blob/main/scripts/server.py
to BattyBirdNET-Analyzer
https://github.com/rdz-oss/BattyBirdNET-Analyzer/blob/main/server.py

Requires a bit of software engineering skill. But would solve the update problem for BirdNET-Pi - could use the latest classifier. Hope that helps,

BR
Richard

**Comment by rdz-oss** [comment]
Hi @gutocarv , hi @Nachtzuster ,

using a custom trained model should now be possible with just a little work in the BattyBirdNEt-Pi fork. It can be used for birds and bats now as well as for your own classifiers trained on top of BattybirdNET-Analyzer which should be analougous to BirdNET-Analyzer. It could be that models trained with BirdNEt-Analyzer are also compatible.
Let me know if you need help. The only thing is that currently you need an audiomoth USB version to run it.

BR
Richard

**Comment by simonlinke** [comment]
Hi Richard - I might get in touch with you soon. Moth is no prob - I
actually just got one of the moth mics.

Simon

On Wed, 15 May 2024 at 05:48, Dr. Richard Zinck ***@***.***>
wrote:

> Hi @gutocarv <https://github.com/gutocarv> , hi @Nachtzuster
> <https://github.com/Nachtzuster> ,
>
> using a custom trained model should now be possible with just a little
> work in the BattyBirdNEt-Pi fork. It can be used for birds and bats now as
> well as for your own classifiers trained on top of BattybirdNET-Analyzer
> which should be analougous to BirdNET-Analyzer. It could be that models
> trained with BirdNEt-Analyzer are also compatible.
> Let me know if you need help. The only thing is that currently you need an
> audiomoth USB version to run it.
>
> BR
> Richard
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Nachtzuster/BirdNET-Pi/discussions/15#discussioncomment-9437965>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AS74J7JK3Y55D3CAOOBF243ZCJTARAVCNFSM6AAAAABEL5ILKCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TIMZXHE3DK>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
-------------
Note that I have recently moved jobs to CSIRO - for work related emails
please contact me here



*Dr Simon Linke *
Senior Research Scientist
Team Leader “Integrated freshwater landscapes”
*CSIRO* Australia’s national science agency

***@***.*** *|* www.csiro.au
Address: 41 Boggo Rd, Dutton Park QLD 4102

**Reply by rdz-oss** [reply]
Cool. Using a fine tuned classifier (bat or bird)  is now supported via the UI. Let me know when you want to give it a try.
